### PR TITLE
DB/UI levels

### PR DESCRIPTION
**Pull request by sderickson**
Add frontend and backend code for enabling DB and UI in Game Dev levels

* Add `keyValueDb` to level sessions
* Set up the client to receive db updates from the game and sync them with the server
* Add `GameDevTrackView` which displays properties specified by players in game dev levels.

### CODE REVIEW

File: app/lib/surface/Surface.coffee [hunk 1]
```diff
@@ -234,6 +234,12 @@ module.exports = Surface = class Surface extends CocoClass
     frame = @world.getFrame(@getCurrentFrame())
     return unless frame
     frame.restoreState()
+
+    if @options.levelType is 'game-dev'
+      Backbone.Mediator.publish('surface:ui-tracked-properties-changed',
```
Comment (sderickson) [inline app/lib/surface/Surface.coffee]: @Zerrien shouldn't this be included in event schemas?

Comment (Zerrien) [inline app/lib/surface/Surface.coffee]: Hm, there are schemas for front-end events?

Comment (sderickson) [inline app/lib/surface/Surface.coffee]: https://github.com/codecombat/codecombat/blob/44596551d7d6b14a59734126391d15c045f5819c/app/schemas/subscriptions/surface.coffee#L176

**Comment by sderickson** [comment]
Add frontend and backend code for enabling DB and UI in Game Dev levels

* Add keyValueDb to level sessions
* Set up the client to receive db updates from the game and sync them with the server
* Add GameDevTrackView which displays properties specified by players in game dev levels.

**Pull request merged**